zebra-consensus/src/block/check.rs

```diff
@@ -168,10 +168,45 @@ pub fn time_is_valid_at(
 /// Check Merkle root validity.
 ///
 /// `transaction_hashes` is a precomputed list of transaction hashes.
+///
+/// # Consensus rules:
+///
+/// - The nConsensusBranchId field MUST match the consensus branch ID used for SIGHASH transaction hashes, as specifed in [ZIP-244] ([7.1]).
+/// - [7.6]
```

Comment:
This field is defined in the block table:
```suggestion
/// - A SHA-256d hash in internal byte order. The merkle root is derived from the hashes of all transactions included in this block, ensuring that none of those transactions can be modified without modifying the header. [7.6]
```

You might also want to wrap these comments at 80 or 100 characters, they're quite long.